File: app/src/main/java/com/gooseco/technicalrulesquiz/data/model/QuizResult.kt
package com.gooseco.technicalrulesquiz.data.model

data class QuizResult(
    val totalQuestions: Int,
    val correctAnswers: Int,
    val incorrectAnswers: Int,
    val answeredQuestions: List<AnsweredQuestion>
) {
    val scorePercentage: Int
        get() = if (totalQuestions > 0) (correctAnswers * 100) / totalQuestions else 0

    val passed: Boolean
        get() = scorePercentage >= 70
}

data class AnsweredQuestion(
    val question: QuizQuestion,
    val userAnswerIndex: Int,
    val isCorrect: Boolean,
    val timeTaken: Long = 0L
)

File: ios/PowerliftingQuiz/Sources/Models/QuizResult.swift
import Foundation

struct AnsweredQuestion {
    let question: QuizQuestion
    let userAnswerIndex: Int
    let isCorrect: Bool
    let timeTaken: TimeInterval
}

struct QuizResult {
    let totalQuestions: Int
    let correctAnswers: Int
    let incorrectAnswers: Int
    let answeredQuestions: [AnsweredQuestion]

    var scorePercentage: Int {
        guard totalQuestions > 0 else { return 0 }
        return (correctAnswers * 100) / totalQuestions
    }

    var passed: Bool { scorePercentage >= 70 }
}
